rwengine/include/engine/GameInputState.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef RWENGINE_GAMEINPUTSTATE_HPP
+#define RWENGINE_GAMEINPUTSTATE_HPP
+
+struct GameInputState
+{
+	/// Inputs that can be controlled
+	/// @todo find any sensible values
+	enum Control {
+		/* On Foot */
+		FireWeapon = 0,
+		NextWeapon,
+		NextTarget = NextWeapon,
+		LastWeapon,
+		LastTarget = LastWeapon,
+		GoForward,
+		GoBackwards,
+		GoLeft,
+		GoRight,
+		ZoomIn,
+		ZoomOut,
+		EnterExitVehicle,
+		ChangeCamera,
+		Jump,
+		Sprint,
+		AimWeapon,
+		Crouch,
+		Walk,
+		LookBehind,
+
+		/* In Vehicle */
+		VehicleFireWeapon = FireWeapon,
+		VehicleAccelerate,
```

Comment:
I was going to consider reversing the buttons pressed from the current player state and logical input:
```
case CROSS:
    if (player->inVehicle())
        return input.pressed(Accelerate);
    else
        return input.pressed(Sprint);
```
Since the current game input handling assumes that those controls are not both activated.

It'd be possible to only track the physical input against the logical state too, just need to update our input handling to take into account the current player state.